lib/src/hitmap.dart

```diff
@@ -35,15 +35,21 @@ class HitMap {
 
   /// Creates a single hitmap from a raw json object.
   ///
+  /// Note that when [checkIgnoredLines] is `true` all files will be
+  /// read to get ignore comments. This will add some overhead.
+  /// To combat this when calling this function multiple times from the
+  /// same source (e.g. test runs of different files) a cache is taken
+  /// via [ignoredLinesInFilesCache]. If this cache contains the parsed
+  /// data for the specific file already, the file will not be read and
+  /// parsed again.
+  ///
   /// Throws away all entries that are not resolvable.
-  static Future<Map<String, HitMap>> parseJson(
+  static Map<String, HitMap> parseJsonSync(
```

Comment:
Why add a separate `parseJsonSync` function? Why not just add a cache to `parseJson`?

Reply:
This - at least I think - would be a non-breaking change.

I think adding it to parseJson (and removing the async) would make it breaking, though I'm quite frankly not sure exactly what constitutes a breaking change.

I've made it sync, btw, to avoid potential race conditions in filling in the cache.

Reply:
I think it's hard to reason about how often a file will be read twice and that's one reason for me to favor the sync version.

As for speed, generally async _can_ be faster when doing multiple io operations simultaneously. If only doing one IO operation at a time though it is slower. `parseJson` only reads one file at a time. Therefore the only way the async version of `parseJson` could be faster would be if it was called several times concurrently. Considering `parseJson` is not optimized for being called several times (at least not if the calls share sources as they for instance do in flutter which is the entire point of the PR) I don't think that's a use-case that should make us favor async over sync.

Said another way, my intuition would certainly be that for the common use case a sync version would be faster than an async version.

Reply:
Ok